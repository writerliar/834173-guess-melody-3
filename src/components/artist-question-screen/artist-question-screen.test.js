import React from "react";
import renderer from "react-test-renderer";
import ArtistQuestionScreen from "./artist-question-screen.jsx";

const question = {
  type: `artist`,
  song: {
    artist: `Ramones`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [
    {
      picture: `https://api.adorable.io/avatars/128/1`,
      artist: `Ramones`,
    },
    {
      picture: `https://api.adorable.io/avatars/128/2`,
      artist: `Nirvana`,
    },
    {
      picture: `https://api.adorable.io/avatars/128/3`,
      artist: `Cranberries`,
    },
  ],
};

it(`Should ArtistQuestionScreen render correctly`, () => {
  const tree = renderer.create((
    <ArtistQuestionScreen
      question={question}
      onAnswer={() => {}}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
