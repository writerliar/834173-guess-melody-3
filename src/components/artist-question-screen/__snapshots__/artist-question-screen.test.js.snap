// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ArtistQuestionScreen render correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Ramones"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        Ramones
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Nirvana"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Nirvana
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Cranberries"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/3"
        />
        Cranberries
      </label>
    </div>
  </form>
</section>
`;
